### DISCUSSION TITLE
Add a tooltip to sidebar list item

### DISCUSSION THREAD

**Discussion by shipswake** [facebook/docusaurus#10529] [General] [answered]
I am generating docs automatically using @graphql-markdown
When the sidebar content is generated some of the labels are quite long
The sidebar is setting the width to fit-content which results in the sidebar width changing as groups are expanded 

![image](https://github.com/user-attachments/assets/56d60546-671b-4d0a-8942-ef4df06073f1)

I would like to:
fix the sidebar width 
truncate the label in the sidebar 
display a tooltip on hover with the complete label

Is this possible?

**Comment by slorber** [comment]
> The sidebar is setting the width to fit-content which results in the sidebar width changing as groups are expanded

![CleanShot 2024-09-27 at 16 40 39](https://github.com/user-attachments/assets/47fa1daf-a1d0-43b8-bef4-8e659f657153)

That's not what I see on our own website. Do you have custom CSS?

Can you please provide a runnable repro of your problem using Docusaurus.new ? We need to be able to see your code and run it under the latest version to help.

**Reply by shipswake** [reply]
yes, it is custom css

**Reply by slorber** [reply]
If you use custom CSS, we can't really do anything for you in that case. 

What we can do instead is to improve our default theme to automatically truncate long sidebar labels and add a title prop.

You can open a bug report so that we don't forget to handle it.

But keep in mind that as soon as you use custom CSS, you diverge from our code and are responsible for restoring the features you eventually break with it.

**Reply by shipswake** [reply]
Yes, I realize the custom CSS takes it out of your hands
When I remove the custom CSS that is where the labels cannot be seen fully

This will work

> What we can do instead is to improve our default theme to automatically truncate long sidebar labels and add a title prop.

**Comment by shipswake** [comment, marked as answer]
I added an issue here: https://github.com/facebook/docusaurus/issues/10531

**Reply by shipswake** [reply]
I know it is a big ask.  Any idea on the timeline for a fix?

**Reply by slorber** [reply]
Unfortunately, I am busy on more important things and don't plan to work on this myself soon. If you submit a PR with a fix, it will be fixed faster. EVentually someone from the community would randomly submit a PR.
